src/fsspec_xrootd/xrootd.py

```diff
@@ -219,6 +321,163 @@ def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
             else:
                 return [os.path.basename(item["name"].rstrip("/")) for item in listing]
 
+    async def _cat_file(self, path: str, start: int, end: int, **kwargs: Any) -> Any:
+        _myFile = client.File()
+        try:
+            status, _n = await _async_wrap(
+                _myFile.open,
+                self.protocol + "://" + self.storage_options["hostid"] + "/" + path,
+                OpenFlags.READ,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"File failed to read: {status.message}")
+            status, data = await _async_wrap(
+                _myFile.read,
+                start,
+                end - start,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"Bytes failed to read from open file: {status.message}")
+            return data
+        except Exception as exc:
+            status, _n = await _async_wrap(
+                _myFile.close,
+                self.timeout,
+            )
+            raise OSError(f"Something went wrong in _cat_file(): {exc}")
```

Comment:
this should just be
```python
    finally:
            status, _n = await _async_wrap(
                _myFile.close,
                self.timeout,
            )
```
so we let the exception raise itself rather than re-raise

Reply:
ok